addons/project/models/project.py

```diff
@@ -1953,6 +1953,14 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
         self._ensure_fields_are_accessible(fields_list)
         return super(Task, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
 
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        if ('personal_stage_type_ids', '=', False) in domain:
+            personal_stages = self.env['project.task.stage.personal'].search([('user_id', '=', self.env.uid)])
+            task_ids = personal_stages.task_id.ids
+            domain = [('id', 'not in', task_ids) if elem == ('personal_stage_type_ids', '=', False) else elem for elem in domain]
```

Comment:
since the user_id (that is used to filter `project.task.type` for field `personal_stage_type_ids`) is stored on the `project.task.type` couldn't we just check that field, eg:

```suggestion
            types = self.env['project.task.type']._search([('user_id', '=', self.env.uid)])                          
            args = [('personal_stage_type_ids', 'not in', types) if leaf == ('personal_stage_type_ids', '=', False) else leaf for leaf in args]
```
but maybe it's different, I'm not sure what `project.task.stage.personal` is.

Reply:
applied :D